server/src/main/java/de/uftos/services/TimetableService.java

```diff
@@ -360,7 +407,12 @@ private List<StudentGroupProblemDto> getStudentGroups(Timetable timetable) {
 
   private List<StudentProblemDto> getStudents() {
     List<StudentProblemDto> students = new ArrayList<>();
-    for (Student student : studentRepository.findAll()) {
+    List<Student> studentList = studentRepository.findAll();
+    if (studentList.isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+          "There must be at least one student to create a timetable");
```

Comment:
Translation required.
Also I'm not so sure about this one...
The timetable has no problems with empty student groups.
I think this one should actually be removed.

Reply:
Removed